tracing-attributes/src/lib.rs

```diff
@@ -973,26 +1137,85 @@ fn path_to_string(path: &Path) -> String {
     res
 }
 
-// A visitor struct replacing the "self" and "Self" tokens in user-supplied fields expressions when
-// the function is generated by async-trait.
-struct SelfReplacer {
-    ty: Option<syn::TypePath>,
+// A visitor struct to replace idents and types in some piece
+// of code (e.g. the "self" and "Self" tokens in user-supplied
+// fields expressions when the function is generated by an old
+// version of async-trait).
+struct KnifeReplacer<'a> {
```

Comment:
why "KnifeReplacer"? that name doesn't really explain what this does, IMO.

Reply:
Yeah, I'm really not good at naming I guess :/
I think when I wrote it I was thinking about swiss knives, but then that code didn't end up as generic or powerful or as I hoped :confounded: 
Not sure the new naming (`IdentAndTypesRenamer`) is much clearer though :)